### PR TITLE
feat(authentication): added stateless authentication

### PR DESCRIPTION
**Pull request by madhavilosetty-intel**

### CODE REVIEW

File: src/app/auth.service.ts [hunk 1]
```diff
@@ -22,6 +22,12 @@ export class AuthService {
     }
   }
 
+  getLoggedUserToken (): any {
```
Comment (rsdmike) [inline src/app/auth.service.ts:25]: string return type

File: src/app/custom.interceptor.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        (data: any) => {
+          if (data instanceof HttpResponse) {
+            return data
+          }
+          return null
+        },
+        (error: any) => {
+          if (error instanceof HttpErrorResponse) {
+            if (error.status === 401) {
+              this.authService.logout()
```
Comment (rsdmike) [inline src/app/custom.interceptor.ts:37]: use `Router` to redirect to /login route

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/app/app.module.ts [hunk 3]
```diff
@@ -49,6 +51,9 @@ import { DeviceToolbarComponent } from './devices/device-toolbar/device-toolbar.
     PowerAlertComponent,
     DeviceToolbarComponent
   ],
+  entryComponents: [
+    DialogcontentComponent
```
Comment (rsdmike) [inline src/app/app.module.ts:55]: change to `DialogContentComponent`

File: src/app/shared/dialogcontent/dialogcontent.component.html [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<mat-dialog-content>{{data.name}}</mat-dialog-content>
+<mat-dialog-actions>
+    <button mat-button mat-dialog-close="false">ok</button>
```
Comment (rsdmike) [inline src/app/shared/dialogcontent/dialogcontent.component.html:3]: lets capital OK here

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/app/app.module.ts (hunk 3 above)
Comment (madhavilosetty-intel) [reply src/app/app.module.ts]: done

File: src/app/shared/dialogcontent/dialogcontent.component.html (hunk 4 above)
Comment (madhavilosetty-intel) [reply src/app/shared/dialogcontent/dialogcontent.component.html]: done

**Review by rsdmike — APPROVED** [review]

**Pull request merged**